nav2_costmap_2d/include/nav2_costmap_2d/layer.hpp

```diff
@@ -68,6 +68,7 @@ class Layer  // TODO(mjeronimo): public nav2_util::LifecycleHelperInterface
   virtual void deactivate() {} /** @brief Stop publishers. */
   virtual void activate() {}   /** @brief Restart publishers if they've been stopped. */
   virtual void reset() = 0;
+  virtual bool isClearable() = 0;
```

Comment:
This seems a little outside the scope of the original PR, but I think the burden of figuring out which layers to clear should stay with the recovery behavior rather than making a flag for it in the layer interface itself.

Reply:
Why? Is this not a property of the layer itself whether it is clearable?

Reply:
I'd rather put the burden on the person configuring the system rather than force every layer-implementer to code for every possible recovery behavior.

Reply:
@DLu @mikeferguson  what are your thoughts here?

tl;dr, changed to add `isClearable()` to layer API so that its a property of the costmap layer rather than a parameter